samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -904,6 +906,19 @@ class SamzaContainer(
     })
   }
 
+  def startDiagnostics {
+    // TODO: where should this reside, MetricConfig? Log4jSystemConfig? or a new separate config?
+    val DIAGNOSTICS_APPENDER_ENABLE = "samza.diagnostics.enabled"
+    if (containerContext.config.getBoolean(DIAGNOSTICS_APPENDER_ENABLE, false)) {
+      import org.apache.log4j.Logger
+      val rootLogger = Logger.getRootLogger
+
+      info("Starting Diagnostics Appender.")
+      val diagnosticsAppender = new DiagnosticsAppender(this.metrics)
+      rootLogger.addAppender(diagnosticsAppender)
```

Comment:
In standalone, there're plans to run more than one `StreamProcessor` in a jvm(which translates to more than one `SamzaContainer` thread in jvm). 
So it will be better to add diagnosticsAppender to rootLogger if it is not already added.